core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UserCoreUtil.java

```diff
@@ -79,6 +76,11 @@
     private static ThreadLocal<UserMgtContext> threadLocalToSetUserMgtContext = new
             ThreadLocal<UserMgtContext>();
 
+    /**
+     * This thread local is used to keep track whether Ask Password option is enabled during the user addition flow.
+     */
+    private static ThreadLocal<Boolean> threadLocalToSetAskPasswordEnabled = new ThreadLocal<>();
```

Comment:
This should be not specific to ask the password. Expectation is to skip password pattern validation if this thread local is set to true. Name needs to be changed.

Reply:
`threadLocalToSkipPasswordPolicyValidation` WDYT?

Reply:
skipPasswordPatternValidation

Reply:
Set to `skipPasswordPatternValidationThreadLocal`.